test/jdk/javax/swing/JRadioButton/ButtonGroupFocus/ButtonGroupFocusTest.java

```diff
@@ -69,52 +100,104 @@ public static void main(String[] args) throws Exception {
+                @Override
+                public void eventDispatched(AWTEvent event) {
+                    System.out.println("Dispatched " + event);
+                }
+            }, AWTEvent.KEY_EVENT_MASK);
+
             frame.pack();
+            frame.setLocationRelativeTo(null);
             frame.setVisible(true);
         });
 
-        robot.waitForIdle();
-        robot.delay(200);
-
-        SwingUtilities.invokeAndWait(() -> {
-            if( !button2.hasFocus() ) {
-                frame.dispose();
-                throw new RuntimeException(
-                        "Button 2 should get focus after activation");
+        try {
+            if (!button2FocusLatch.await(FOCUS_TIMEOUT, SECONDS)) {
+                throw new RuntimeException("Button 2 should get focus "
+                                           + "after activation");
             }
-        });
+            robot.waitForIdle();
+            robot.delay(200);
 
-        robot.keyPress(KeyEvent.VK_TAB);
-        robot.keyRelease(KeyEvent.VK_TAB);
+            System.out.println("\n\n*** Tab 1st");
+            robot.keyPress(KeyEvent.VK_TAB);
+            robot.keyRelease(KeyEvent.VK_TAB);
 
-        robot.waitForIdle();
-        robot.delay(200);
+            if (!button4FocusLatch.await(FOCUS_TIMEOUT, SECONDS)) {
+                throw new RuntimeException("Button 4 should get focus");
+            }
+            robot.waitForIdle();
+            robot.delay(200);
 
-        SwingUtilities.invokeAndWait(() -> {
-            if( !button4.hasFocus() ) {
-                frame.dispose();
-                throw new RuntimeException(
-                        "Button 4 should get focus");
+            if (button2FocusLatch2.await(1, MILLISECONDS)) {
+                throw new RuntimeException("Focus moved back to Button 2");
             }
-            button3.setSelected(true);
-        });
 
-        robot.keyPress(KeyEvent.VK_TAB);
-        robot.keyRelease(KeyEvent.VK_TAB);
+            SwingUtilities.invokeAndWait(() -> button3.setSelected(true));
+            robot.waitForIdle();
+            robot.delay(200);
 
-        robot.waitForIdle();
-        robot.delay(200);
+            System.out.println("\n\n*** Tab 2nd");
+            robot.keyPress(KeyEvent.VK_TAB);
+            robot.keyRelease(KeyEvent.VK_TAB);
 
-        SwingUtilities.invokeAndWait(() -> {
-            if( !button3.hasFocus() ) {
-                frame.dispose();
-                throw new RuntimeException(
-                        "selected Button 3 should get focus");
+            if (!button3FocusLatch.await(FOCUS_TIMEOUT, SECONDS)) {
+                throw new RuntimeException("Selected Button 3 should get focus");
             }
-        });
+        } catch (Exception e) {
+            BufferedImage image = robot.createScreenCapture(getFrameBounds());
+            ImageIO.write(image, "png",
+                          new File("image.png"));
```

Comment:
Does this overflow the 80 character line limit? Or is this a standard?

Reply:
Yes, it does… by two characters. I find this way easier to parse.